Gemfile

```diff
@@ -4,3 +4,7 @@ source "http://rubygems.org"
 gemspec
 
 gem 'rubysl', '~> 2.0', :platform => :rbx
+
+if RUBY_VERSION < '2.1.0'
+  gem 'nokogiri', '~> 1.6.8'
+end
```

Comment:
I really appreciate you following up on this.  I am sorry to keep nit-picking your contribution, but I don't think this works when another project uses approvals as a gem.  In that case, the other project will only see what's in the `approvals.gemspec` file and if that project uses Ruby < 2.1, it would pull in the wrong version of nokogiri unless it lists it as a direct dependency.

We had to solve this issue before with the `json` gem.  See https://github.com/kytrinyx/approvals/blob/master/approvals.gemspec#L27-L28 for the comment that mentions it and https://github.com/kytrinyx/approvals/blob/master/ext/mkrf_conf.rb#L9-L17 for the implementation.

Reply:
> I don't think this works when another project uses approvals as a gem.

A fair point. Apologies.

Reply:
On further consideration, there are two issues related to installing the appropriate version of Nokogiri here:
1. Installing gems for the CI job. Changing `ext/mkrf_conf.rb` _will not_ fix this.
1. The gem installation itself. Changing `ext/mkrf_conf.rb` _will_ fix this.

Given that the `approvals.gemspec` itself is Ruby code and is evaluated as such, I think it will make more sense to put control flow for evaluating an appropriate `nokogiri` version there.

I'll give it a go now.

Reply:
That approach worked. 🎉

Reply:
I've tested this locally using Docker images for different Ruby versions. The specification file which was generated for Ruby 2.4.1:
```
/tmp # cat /usr/local/bundle/specifications/approvals-0.0.24.gemspec 
# -*- encoding: utf-8 -*-
# stub: approvals 0.0.24 ruby lib
# stub: ext/mkrf_conf.rb

Gem::Specification.new do |s|
  s.name = "approvals".freeze
  s.version = "0.0.24"

  s.required_rubygems_version = Gem::Requirement.new("> 1.3.1".freeze) if s.respond_to? :required_rubygems_version=
  s.require_paths = ["lib".freeze]
  s.authors = ["Katrina Owen".freeze]
  s.date = "2018-01-18"
  s.description = "A library to make it easier to do golden-master style testing in Ruby".freeze
  s.email = ["katrina.owen@gmail.com".freeze]
  s.executables = ["approvals".freeze]
  s.extensions = ["ext/mkrf_conf.rb".freeze]
  s.files = ["bin/approvals".freeze, "ext/mkrf_conf.rb".freeze]
  s.homepage = "https://github.com/kytrinyx/approvals".freeze
  s.licenses = ["MIT".freeze]
  s.rubyforge_project = "approvals".freeze
  s.rubygems_version = "2.6.12".freeze
  s.summary = "Approval Tests for Ruby".freeze

  s.installed_by_version = "2.6.12" if s.respond_to? :installed_by_version

  if s.respond_to? :specification_version then
    s.specification_version = 4

    if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
      s.add_development_dependency(%q<rspec>.freeze, ["~> 3.1"])
      s.add_runtime_dependency(%q<thor>.freeze, ["~> 0.18"])
      s.add_runtime_dependency(%q<nokogiri>.freeze, ["~> 1.8"])
    else
      s.add_dependency(%q<rspec>.freeze, ["~> 3.1"])
      s.add_dependency(%q<thor>.freeze, ["~> 0.18"])
      s.add_dependency(%q<nokogiri>.freeze, ["~> 1.8"])
    end
  else
    s.add_dependency(%q<rspec>.freeze, ["~> 3.1"])
    s.add_dependency(%q<thor>.freeze, ["~> 0.18"])
    s.add_dependency(%q<nokogiri>.freeze, ["~> 1.8"])
  end
end
```

Reply:
And the same for Ruby 1.9:
```
$ cat /usr/local/bundle/specifications/approvals-0.0.24.gemspec 
# -*- encoding: utf-8 -*-

Gem::Specification.new do |s|
  s.name = "approvals"
  s.version = "0.0.24"

  s.required_rubygems_version = Gem::Requirement.new("> 1.3.1") if s.respond_to? :required_rubygems_version=
  s.authors = ["Katrina Owen"]
  s.date = "2018-01-18"
  s.description = "A library to make it easier to do golden-master style testing in Ruby"
  s.email = ["katrina.owen@gmail.com"]
  s.executables = ["approvals"]
  s.extensions = ["ext/mkrf_conf.rb"]
  s.files = ["bin/approvals", "ext/mkrf_conf.rb"]
  s.homepage = "https://github.com/kytrinyx/approvals"
  s.licenses = ["MIT"]
  s.require_paths = ["lib"]
  s.rubyforge_project = "approvals"
  s.rubygems_version = "1.8.23.2"
  s.summary = "Approval Tests for Ruby"

  if s.respond_to? :specification_version then
    s.specification_version = 3

    if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
      s.add_development_dependency(%q<rspec>, ["~> 3.1"])
      s.add_runtime_dependency(%q<thor>, ["~> 0.18"])
      s.add_runtime_dependency(%q<nokogiri>, ["~> 1.6.8"])
    else
      s.add_dependency(%q<rspec>, ["~> 3.1"])
      s.add_dependency(%q<thor>, ["~> 0.18"])
      s.add_dependency(%q<nokogiri>, ["~> 1.6.8"])
    end
  else
    s.add_dependency(%q<rspec>, ["~> 3.1"])
    s.add_dependency(%q<thor>, ["~> 0.18"])
    s.add_dependency(%q<nokogiri>, ["~> 1.6.8"])
  end
end
```